x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveReasonTests.java

```diff
@@ -37,12 +44,21 @@ public void testXContent() throws IOException {
         String indexName = randomAlphaOfLength(10);
         SortedSet<ShardId> unassignedShardIds = new TreeSet<>(randomUnique(() -> new ShardId(indexName, indexUUID, randomInt(1000)), 600));
         SortedSet<ShardId> assignedShardIds = new TreeSet<>(randomUnique(() -> new ShardId(indexName, indexUUID, randomInt(1000)), 600));
+        DiscoveryNode discoveryNode = new DiscoveryNode("node1", buildNewFakeTransportAddress(), emptyMap(), emptySet(), Version.CURRENT);
+        List<NodeAllocationResult> unassignedShardAllocationDecision = List.of(
+            new NodeAllocationResult(discoveryNode, AllocationDecision.NO, Decision.NO, 1)
```

Comment:
Can we add a Decision that includes more details using `org.elasticsearch.cluster.routing.allocation.decider.Decision#single`?